src/routes/profile/index.tsx

```diff
@@ -66,13 +68,44 @@ router.get('/avatargetlinktutorial', (req, res) => {
 
 // Show the mod approving rejecting page
 router.get('/mod/customavatar', isModMiddleware, async (req, res) => {
-  const customAvatarRequests = await avatarRequest.find({ processed: false });
   const avatarLookupReact = renderToString(<AvatarLookup />);
+  const customAvatarRequests = await avatarRequest.find({ processed: false });
+
+  interface UpdatedAvatarRequest {
+    id: string;
+    forUsername: string;
+    resLink: string;
+    spyLink: string;
+    lastApprovedAvatarDate: Date | null;
+    enoughTimeElapsed: boolean; // TODO-kev: Think of a better name
+  }
+
+  const updatedAvatarRequests: Promise<UpdatedAvatarRequest[]> =
+    await Promise.all(
+      customAvatarRequests.map(async (avatarReq) => {
+        const user = await userAdapter.getUser(avatarReq.forUsername);
+
+        return {
+          id: avatarReq.id,
+          forUsername: avatarReq.forUsername,
+          resLink: avatarReq.resLink,
+          spyLink: avatarReq.spyLink,
+          lastApprovedAvatarDate: user.lastApprovedAvatarDate
+            ? user.lastApprovedAvatarDate
+            : null,
+          enoughTimeElapsed: user.lastApprovedAvatarDate
+            ? user.lastApprovedAvatarDate <
+              new Date() - MIN_TIME_SINCE_LAST_AVATAR_APPROVAL
+            : true,
```

Comment:
Add in library size check (should not show red to mods if they are currently at 1 but they have lib size 2). And think of a better name.